source/common/router/upstream_request.cc

```diff
@@ -486,6 +486,10 @@ void UpstreamRequest::onPoolReady(
 
   if (span_ != nullptr) {
     span_->injectContext(*parent_.downstreamHeaders());
+  } else {
+    // No independent child span for current upstream request then inject the parent span's tracing
+    // context into the request headers.
```

Comment:
OK I understand now, thanks for explaining. Do you mind adding a few more comments here on the implications of doing this multiple times for retries, etc.? I understand this is the way it was always done, but it's a little more confusing now. Thank you!

/wait

Reply:
get it.